tensorflow/python/distribute/input_lib_test.py

```diff
@@ -1194,5 +1194,45 @@ def dataset_fn(ctx):
           strategy,
           sess=sess)
 
+class DistributedIteratorPerReplicaTest(test.TestCase, parameterized.TestCase):
+
+  @combinations.generate(
+      combinations.combine(
+          mode=["eager"],
+          distribution=[
+              strategy_combinations.mirrored_strategy_with_two_gpus]))
```

Comment:
can we also add `mirrored_strategy_with_cpu_1_and_2` so that it gets tested on CPUs as well? 
And also `mirrored_strategy_with_gpu_and_cpu` which will get tested on 1 GPU machine.

(because I think the github presubmit does not have 2 GPUs)

Note that you will need to add `set_virtual_cpus_to_at_least(3)` in your test's setUp methods.

Reply:
This is addressed for now, but I think it might need to get changed as the results of discussions around https://github.com/tensorflow/tensorflow/pull/38968#discussion_r486760778